velox/exec/ExchangeSource.h

```diff
@@ -74,19 +74,38 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   };
 
   /// Requests the producer to generate up to 'maxBytes' more data and reply
-  /// within 'maxWaitSeconds'. Returns a future that completes when producer
-  /// responds either with 'data' or with a message indicating that all data has
-  /// been already produced or data will take more time to produce.
+  /// within 'maxWait'. Returns a future that completes when producer responds
+  /// either with 'data' or with a message indicating that all data has been
+  /// already produced or data will take more time to produce.
   virtual folly::SemiFuture<Response> request(
       uint32_t maxBytes,
-      uint32_t maxWaitSeconds) = 0;
+      std::chrono::nanoseconds maxWait) {
+    auto maxWaitSeconds = std::max(1ll, llround(maxWait.count() / 1e9));
+    return request(maxBytes, maxWaitSeconds);
+  }
 
   /// Ask for available data sizes that can be fetched.  Normally should not
   /// fetching any actual data (i.e. Response::bytes should be 0).  However for
   /// backward compatibility (e.g. communicating with coordinator), we allow
   /// small data (1MB) to be returned.
   virtual folly::SemiFuture<Response> requestDataSizes(
-      uint32_t maxWaitSeconds) = 0;
+      std::chrono::nanoseconds maxWait) {
+    auto maxWaitSeconds = std::max(1ll, llround(maxWait.count() / 1e9));
+    return requestDataSizes(maxWaitSeconds);
+  }
+
+  /// Used in presto_cpp for backward compatibility only, will be removed soon.
```

Comment:
Can we wrap it in ENABLE_BACKWARD_... macro?

Reply:
No we cannot because we still need to forward the definition from the new ones to these 2 for now